src/main/java/com/salesforce/storm/spout/dynamic/metrics/Metrics.java

```diff
@@ -0,0 +1,252 @@
+package com.salesforce.storm.spout.dynamic.metrics;
+
+import com.salesforce.storm.spout.dynamic.metrics.annotation.Documentation;
+import com.sun.scenario.effect.Offset;
+import org.apache.storm.shade.com.codahale.metrics.Gauge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Enumeration of all available metrics.
+ */
+public enum Metrics implements MetricDefinition {
+
+    /*
+     * Spout Monitor Metrics.
+     */
+    @Documentation(
+        type = Documentation.Type.GAUGE,
```

Comment:
I feel like we should probably use a different class for the type and unit attributes here.

I'd also be in favor of creating a distinct annotation for these.